src/condor_utils/file_transfer.cpp

```diff
@@ -6435,15 +6468,68 @@ FileTransfer::ExpandFileTransferList( char const *src_path, char const *dest_dir
 			// dprintf( D_ALWAYS, ">>> not preserving relative path.\n" );
 			destination += condor_basename(src_path);
 		} else {
-			// dprintf( D_ALWAYS, ">>> preserving relative path.\n" );
-			destination += src_path;
+			if(! fullpath(src_path)) {
+				// dprintf( D_ALWAYS, ">>> preserving relative path of relative path %s\n", src_path );
 
-			// ExpandParentDirectories() adds this back in the correct place.
-			expanded_list.pop_back();
+				if( destination.length() > 0 ) { destination += DIR_DELIM_CHAR; }
+				destination += src_path;
+
+				// ExpandParentDirectories() adds this back in the correct place.
+				expanded_list.pop_back();
+
+				if(! ExpandParentDirectories( src_path, iwd, expanded_list, SpoolSpace )) {
+					return false;
+				}
+			} else {
+				//
+				// The only absolute paths we want to treat as relative paths
+				// are absolute paths into the SPOOL directory.  Everything
+				// else should be transferred as usual.
+				//
+
+				ASSERT( SpoolSpace == NULL || fullpath(SpoolSpace) );
+				if( SpoolSpace != NULL && starts_with(src_path, SpoolSpace) ) {
+					const char * relative_path = &src_path[strlen(SpoolSpace)];
+					if( IS_ANY_DIR_DELIM_CHAR(relative_path[0]) ) { ++relative_path; }
+					// dprintf( D_ALWAYS, ">>> preserving relative path of directory (%s) in SPOOL (as %s)\n", src_path, relative_path );
+
+					//
+					// relative_path is relative to SpoolSpace, not the
+					// destination, which means we can't use it to set
+					// destination.  Instead, see if destination is a prefix
+					// to relative_path, and correct if it is.
+					//
+					ASSERT(! fullpath(destination.c_str()));
```

Comment:
Is there no user control over `destination` here?  Again, `ASSERT` seems potentially harsh.

Reply:
[Sorry for the delayed reply here; I wrote up a reply and apparently lost it somewhere along the way.]

There is no user control over `destination`.  Also, see previous comment about ASSERT.